test/unittests/test_dd_synthesis.cpp

```diff
@@ -0,0 +1,551 @@
+        c011101 = TruthTable::Cube::fromInteger(0b011101U, 6U);
+        c011110 = TruthTable::Cube::fromInteger(0b011110U, 6U);
+        c011111 = TruthTable::Cube::fromInteger(0b011111U, 6U);
+
+        c100000 = TruthTable::Cube::fromInteger(0b100000U, 6U);
+        c100001 = TruthTable::Cube::fromInteger(0b100001U, 6U);
+        c100010 = TruthTable::Cube::fromInteger(0b100010U, 6U);
+        c100011 = TruthTable::Cube::fromInteger(0b100011U, 6U);
+        c100100 = TruthTable::Cube::fromInteger(0b100100U, 6U);
+        c100101 = TruthTable::Cube::fromInteger(0b100101U, 6U);
+        c100110 = TruthTable::Cube::fromInteger(0b100110U, 6U);
+        c100111 = TruthTable::Cube::fromInteger(0b100111U, 6U);
+        c101000 = TruthTable::Cube::fromInteger(0b101000U, 6U);
+        c101001 = TruthTable::Cube::fromInteger(0b101001U, 6U);
+        c101010 = TruthTable::Cube::fromInteger(0b101010U, 6U);
+        c101011 = TruthTable::Cube::fromInteger(0b101011U, 6U);
+        c101100 = TruthTable::Cube::fromInteger(0b101100U, 6U);
+        c101101 = TruthTable::Cube::fromInteger(0b101101U, 6U);
+        c101110 = TruthTable::Cube::fromInteger(0b101110U, 6U);
+        c101111 = TruthTable::Cube::fromInteger(0b101111U, 6U);
+
+        c110000 = TruthTable::Cube::fromInteger(0b110000U, 6U);
+        c110001 = TruthTable::Cube::fromInteger(0b110001U, 6U);
+        c110010 = TruthTable::Cube::fromInteger(0b110010U, 6U);
+        c110011 = TruthTable::Cube::fromInteger(0b110011U, 6U);
+        c110100 = TruthTable::Cube::fromInteger(0b110100U, 6U);
+        c110101 = TruthTable::Cube::fromInteger(0b110101U, 6U);
+        c110110 = TruthTable::Cube::fromInteger(0b110110U, 6U);
+        c110111 = TruthTable::Cube::fromInteger(0b110111U, 6U);
+        c111000 = TruthTable::Cube::fromInteger(0b111000U, 6U);
+        c111001 = TruthTable::Cube::fromInteger(0b111001U, 6U);
+        c111010 = TruthTable::Cube::fromInteger(0b111010U, 6U);
+        c111011 = TruthTable::Cube::fromInteger(0b111011U, 6U);
+        c111100 = TruthTable::Cube::fromInteger(0b111100U, 6U);
+        c111101 = TruthTable::Cube::fromInteger(0b111101U, 6U);
+        c111110 = TruthTable::Cube::fromInteger(0b111110U, 6U);
+        c111111 = TruthTable::Cube::fromInteger(0b111111U, 6U);
+    }
+};
+
+TEST_F(DDSynth, CUSTOMTT1) {
+    tt.insert(c001, c001);
+    tt.insert(c010, c100);
+    tt.insert(c011, c101);
+    tt.insert(c100, c111);
+    tt.insert(c101, c010);
+    tt.insert(c110, c011);
+    tt.insert(c111, c110);
+
+    EXPECT_EQ(tt.ioCube().size(), 7U);
+
+    tt.extend();
+
+    EXPECT_EQ(tt.ioCube().size(), 8U);
+
+    EXPECT_EQ(tt.nInputs(), 3U);
+    EXPECT_EQ(tt.nOutputs(), 3U);
+
+    const auto ttDD = tt.buildDD(dd3Bit);
+    EXPECT_TRUE(ttDD.p != nullptr);
+
+    EXPECT_NE(ttDD, dd3Bit->makeIdent(3U));
+
+    auto src = ttDD;
+
+    EXPECT_NE(src, dd3Bit->makeIdent(3U));
+
+    algoQ(src, dd3Bit);
+
+    EXPECT_EQ(src, dd3Bit->makeIdent(3U));
```

Comment:
while it is good to know that the synthesis is able to transform the DD to the identity, this alone is not sufficient for ensuring that the synthesis works as intended. For this, the resulting circuit has to be checked. Adjust these tests accordingly.